### PR TITLE
A few evolution updates

### PR DESCRIPTION
**Pull request by LOuroboros**
## Description
- Changed the Eevee->Leafeon evo method from EVO_MAP to EVO_ITEM, which is more accurate in comparison as of Gen. 8 since we still don't have the former evo method properly implemented.
- Changed the Eevee->Glaceon evo method for the same reason.
- Added Charjabug's Thunder Stone based evolution as a second option.

I didn't remove the EVO_MAP based evo from Charjabug because it's an accurate way of representing magnetic field reliant evolutions.

## **Discord contact info**
Lunos#4026

### CODE REVIEW

**Comment by LOuroboros** [comment]
Like Charjabug, Magneton can also evolve with a Thunder Stone now, so I threw that one in too.

**Comment by AsparagusEduardo** [comment]
I don't get this change. Mossy Rock and Icy Rock evolutions are also map-based (you can evolve anywhere in the map the rock is in.
You could add the stone evolutions for Leafeon and Glaceon, but I don't see why replace them.

File: src/data/pokemon/evolution.h [hunk 1]
```diff
@@ -99,8 +100,8 @@ const struct Evolution gEvolutionTable[NUM_SPECIES][EVOS_PER_MON] =
                                        {EVO_ITEM, ITEM_NONE,/*ITEM_FIRE_STONE*/ SPECIES_FLAREON},
                                        {EVO_FRIENDSHIP_DAY, 0, SPECIES_ESPEON},
                                        {EVO_FRIENDSHIP_NIGHT, 0, SPECIES_UMBREON},
-                                       {EVO_MAP, MAPSEC_PETALBURG_WOODS, SPECIES_LEAFEON},
-                                       {EVO_MAP, MAPSEC_SHOAL_CAVE, SPECIES_GLACEON},
```
Comment (ExpoSeed) [inline src/data/pokemon/evolution.h:103]: no need to remove, they can coexist like they do for charjabug and magneton

**Comment by LOuroboros** [comment]
> You could add the stone evolutions for Leafeon and Glaceon, but I don't see why replace them.

Mostly because the current max. amount of evos is set to 8. If that can be safely changed though, I can throw that in here.

File: src/data/pokemon/evolution.h (hunk 1 above)
Comment (ExpoSeed) [reply src/data/pokemon/evolution.h:103]: oops didn't consider max evos

**Comment by LOuroboros** [comment]
Ready for re-review :eyes:

**Comment by LOuroboros** [comment]
Ready for re-re-review 👀

**Review by ExpoSeed — APPROVED** [review]

File: src/data/pokemon/evolution.h [hunk 2]
```diff
@@ -425,9 +425,9 @@ const struct Evolution gEvolutionTable[NUM_SPECIES][EVOS_PER_MON] =
     [SPECIES_TRUMBEAK]              = {{EVO_LEVEL, 28, SPECIES_TOUCANNON}},
     [SPECIES_YUNGOOS]               = {{EVO_LEVEL_DAY, 20, SPECIES_GUMSHOOS}},
     [SPECIES_GRUBBIN]               = {{EVO_LEVEL, 20, SPECIES_CHARJABUG}},
-    [SPECIES_CHARJABUG]             = {{EVO_MAP, MAPSEC_NEW_MAUVILLE, SPECIES_VIKAVOLT},
+    [SPECIES_CHARJABUG]             = {{EVO_MAPSEC, MAPSEC_NEW_MAUVILLE, SPECIES_VIKAVOLT},
                                        {EVO_ITEM, ITEM_NONE,/*ITEM_THUNDER_STONE*/ SPECIES_VIKAVOLT}},
-    [SPECIES_CRABRAWLER]            = {{EVO_MAP, 67, SPECIES_CRABOMINABLE}},
+    [SPECIES_CRABRAWLER]            = {{EVO_MAPSEC, 67, SPECIES_CRABOMINABLE}},
```
Comment (AsparagusEduardo) [inline src/data/pokemon/evolution.h:430]: Crabrawler's evo needs to be updated to Shoal Cave.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Review by AsparagusEduardo — APPROVED** [review]

**Review by ExpoSeed — APPROVED** [review]

**Pull request merged**